code/modules/mob/living/carbon/human/species_types/flypeople.dm

```diff
@@ -138,14 +138,13 @@
 	name = odd_organ_name()
 	icon_state = pick("brain-x-d", "liver-x", "kidneys-x", "spinner-x", "lungs-x", "random_fly_1", "random_fly_2", "random_fly_3", "random_fly_4", "random_fly_5")
 
-/obj/item/organ/internal/stomach/fly/on_life(delta_time, times_fired)
-	if(locate(/datum/reagent/consumable) in reagents.reagent_list)
-		var/mob/living/carbon/body = owner
-		// we do not loss any nutrition as a fly when vomiting out food
-		body.vomit(0, FALSE, FALSE, 2, TRUE, force=TRUE, purge_ratio = 0.67)
-		playsound(get_turf(owner), 'sound/effects/splat.ogg', 50, TRUE)
-		body.visible_message(span_danger("[body] vomits on the floor!"), \
-					span_userdanger("You throw up on the floor!"))
+/obj/item/organ/internal/stomach/fly/after_eat(edible)
+	var/mob/living/carbon/body = owner
+	// we do not loss any nutrition as a fly when vomiting out food
+	body.vomit(0, FALSE, FALSE, 2, TRUE, force=TRUE, purge_ratio = 0.67)
```

Comment:
I know you didn't write this, but please write out named parameters for each of these, it is not obvious what any of them are.